File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/ServerOrderMemory.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ServerOrderMemory._

case class ServerOrderMemory (
  /* Memory ID. */
  id: Option[String],
/* Memory price. */
  price: Option[String],
/* Memory image. */
  img: Option[String],
/* Short description of the memory. */
  short_desc: Option[String],
/* Long description of the memory. */
  long_desc: Option[String],
/* Manufacturer information. */
  manu: Option[String],
/* Memory size. */
  size: Option[String],
/* Memory type. */
  `type`: Option[String],
/* Hidden status. */
  hidden: Option[String],
/* Monthly price. */
  monthly_price: Option[Integer],
/* Drive type. */
  drive_type: Option[String],
/* Display of monthly memory price. */
  monthly_price_display: Option[String])

object ServerOrderMemory {
  import DateTimeCodecs._

  implicit val ServerOrderMemoryCodecJson: CodecJson[ServerOrderMemory] = CodecJson.derive[ServerOrderMemory]
  implicit val ServerOrderMemoryDecoder: EntityDecoder[ServerOrderMemory] = jsonOf[ServerOrderMemory]
  implicit val ServerOrderMemoryEncoder: EntityEncoder[ServerOrderMemory] = jsonEncoderOf[ServerOrderMemory]
}

File: OPENAPI-REST-API/openapi-client/clojure/src/inter_server_management_api/specs/server_order_memory.clj
(ns inter-server-management-api.specs.server-order-memory
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def server-order-memory-data
  {
   (ds/opt :id) string?
   (ds/opt :price) string?
   (ds/opt :img) string?
   (ds/opt :short_desc) string?
   (ds/opt :long_desc) string?
   (ds/opt :manu) string?
   (ds/opt :size) string?
   (ds/opt :type) string?
   (ds/opt :hidden) string?
   (ds/opt :monthly_price) int?
   (ds/opt :drive_type) string?
   (ds/opt :monthly_price_display) string?
   })

(def server-order-memory-spec
  (ds/spec
    {:name ::server-order-memory
     :spec server-order-memory-data}))
